animations/base_clip.py

```diff
@@ -0,0 +1,92 @@
+from moviepy.editor import *
+
+
+class BaseClip:
+
+    def __init__(self):
+        self.clip = None
+
+    def add_text(self, text, font_size, color, font, interline, pos, duration):
+        text = TextClip(text, fontsize=font_size, color=color,
+                        font=font, interline=interline).set_pos(pos).set_duration(duration)
+        self.clip = CompositeVideoClip([self.clip, text])
```

Comment:
The issue with this line is, once you turn self.clip into CompositeVideoClip, you can't use any base clip specific methods unless they're for clip or CompositeVideoClip. We need to make sure that's the workflow.

Reply:
I was able to add text, crop and add text again without any issues